src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -505,21 +505,14 @@ paths:
           in: query
           description: The field to use for sorting.
           schema:
-            type: string
             default: created_date
-            enum:
-              - name
-              - description
-              - created_date
+            $ref: '#/components/schemas/EnumerateSortByParam'
         - name: direction
           in: query
           description: The direction to sort.
           schema:
-            type: string
-            default: desc
-            enum:
-              - asc
-              - desc
+            default: asc
```

Comment:
Why are we changing the default sort order?

Reply:
We're not. The default sort order has always been `asc`. https://github.com/DataBiosphere/jade-data-repo/blob/0e571545d5ad7530831ebde1c60fe5856810f79c/src/main/java/bio/terra/app/controller/RepositoryApiController.java#L182

Reply:
Oh I see! Thanks for clarifying that.

Reply:
This seems wrong, we shouldn't be specifying a default in two different places. Most likely these annotations were copied to the implementation code when they were first implemented and weren't updated when the yaml changed. So right now the yaml says one thing and the annotation in the implementation says another. The generated code in `RepositoryApi` matches the yaml. Not sure which one is used at runtime if they disagree.

My hope is that we can safely delete all the `RequestParam` annotations in the overridden methods to avoid future divergence, and rely only on the yaml and the openapi generated code.

Reply:
It looks like the interface code doesn't specify a default, which is interesting since there's a default specified in the yaml. I wonder if that's not a supported feature in Java.

Reply:
It looks like the default value in the OpenAPI is the default option that is selected in the swagger UI. But if you don't choose an option, you get a list of datasets/snapshots in ascending order.

Reply:
> It looks like the interface code doesn't specify a default, which is interesting since there's a default specified in the yaml. I wonder if that's not a supported feature in Java.

That's odd, I can see it in my generated copy of `RepsitoryApi.java`.

```
    default ResponseEntity<EnumerateSnapshotModel> enumerateSnapshots(@ApiParam(value = "The number of items to skip before starting to collect the result set.", defaultValue = "0") @Valid @RequestParam(value = "offset", required = false, defaultValue="0") Integer offset
,@ApiParam(value = "The numbers of items to return.", defaultValue = "10") @Valid @RequestParam(value = "limit", required = false, defaultValue="10") Integer limit
,@ApiParam(value = "The field to use for sorting.", allowableValues = "name, description, created_date", defaultValue = "created_date") @Valid @RequestParam(value = "sort", required = false, defaultValue="created_date") String sort
,@ApiParam(value = "The direction to sort.", allowableValues = "asc, desc", defaultValue = "desc") @Valid @RequestParam(value = "direction", required = false, defaultValue="desc") String direction
,@ApiParam(value = "Filter the results where this string is a case insensitive match in the name or description.") @Valid @RequestParam(value = "filter", required = false) String filter
,@ApiParam(value = "Filter the results where these datasetIds are source datasets.") @Valid @RequestParam(value = "datasetIds", required = false) List<String> datasetIds
) {
```